scipy/sparse/tests/test_base.py

```diff
@@ -1653,22 +1683,42 @@ def test_pow(self):
         A = array([[1, 0, 2, 0], [0, 3, 4, 0], [0, 5, 0, 0], [0, 6, 7, 8]])
         B = self.spcreator(A)
 
-        for exponent in [0,1,2,3]:
-            ret_sp = B**exponent
-            ret_np = np.linalg.matrix_power(A, exponent)
-            assert_array_equal(ret_sp.toarray(), ret_np)
-            assert_equal(ret_sp.dtype, ret_np.dtype)
-
-        # invalid exponents
-        for exponent in [-1, 2.2, 1 + 3j]:
-            assert_raises(ValueError, B.__pow__, exponent)
-
-        # nonsquare matrix
-        B = self.spcreator(A[:3,:])
-        assert_raises(TypeError, B.__pow__, 1)
+        if self.is_array_test:
+            # Todo: Add test of B**(1+3j) when np1.24 is no longer supported.
```

Comment:
Not sure this comment can be interpreted outside the context of this diff. Looks like 3 needs to be changed to 1+3j in a future version?

Reply:
I intended it to mean that 1+3j should be added to the list of exponents when np1.24 is no longer supported. I'll also update the comment to make it clear that this is element-wise (because array) and that the trouble is that 0 is the fill-value and complex exponents of that create trouble for np1.24.